instaloader/structures.py

```diff
@@ -377,20 +377,23 @@ def is_video(self) -> bool:
     def video_url(self) -> Optional[str]:
         """URL of the video, or None."""
         if self.is_video:
-            version_urls = [self._field('video_url')]
+            version_urls = []
+            try:
+                version_urls.append(self._field('video_url'))
+            except (InstaloaderException, KeyError, IndexError) as err:
+                self._context.error(f"Warning: Unable to fetch video from graphql of {self}: {err}")
             if self._context.iphone_support and self._context.is_logged_in:
                 try:
                     version_urls.extend(version['url'] for version in self._iphone_struct['video_versions'])
                 except (InstaloaderException, KeyError, IndexError) as err:
                     self._context.error(f"Unable to fetch high-quality video version of {self}: {err}")
-                    return version_urls[0]
-            else:
+            version_urls = list(dict.fromkeys(version_urls))
```

Comment:
This is to remove duplicates, right? It might be a little cleaner to use a `set` for `version_urls` rather than a list. It does not store the order of items, so we will loose the enumeration in the warning message "Video URL candidate 1/4 for ..."; but neither does `list(dict.fromkeys(...))` ensure that the order is kept. But that's okay anyway, there is probably little value in printing the index of the URL that fails.

Reply:
Yeah, this is to remove duplicates. I prefer this way than `set` exactly because of the order preservation (even though not very important here), and IIRC since Py 3.5 dicts are ordered, so this operation will keep the order of items.

Reply:
Ah interesting, did not know that. Then it is fine. Thanks!